### PR TITLE
[Phase 1] Refactor Testing-Code-Quality.md

### LINKED ISSUE #2212: [Phase 1] Refactor 0x04h-Testing-Code-Quality.md
1. Go through each test case (1 MR per chapter) 
2. Extract any general information and put it to the theory overview

  - Each chapter must have a “## Overview” section and then several “## Testing …(MSTG-XXX-YY)” sections

  - Strategy 1: move theory to chapter overview
  - Strategy 2: move theory to general chapter overview

3. Only leave a test description on each test (see notes)

Notes:

- Follow the [style guide](https://mas.owasp.org/contributing/5_Style_Guide/)
- If it helps you may already “pre-split” some tests by using:
  - “### “ subsections ([example](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x04i-Testing-User-Privacy-Protection.md#testing-user-education-mstg-storage-12))
  - or “-” bullet point lists if there’s not enough details yet ([example](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05d-Testing-Data-Storage.md#testing-local-storage-for-input-validation-mstg-platform-2), right before the linked test there’s a list with several “- Check …”).

### PR DESCRIPTION
**Pull request by cpholguera**
This PR closes #2212, #2218 and #2225

### CODE REVIEW

File: Document/0x04h-Testing-Code-Quality.md [hunk 1]
```diff
@@ -91,15 +91,15 @@ Verify that the following best practices have been followed:
...
 
 In the context of _native apps_, XSS risks are far less prevalent for the simple reason these kinds of applications do not rely on a web browser. However, apps using WebView components, such as `WKWebView` or the deprecated `UIWebView` on iOS and `WebView` on Android, are potentially vulnerable to such attacks.
 
 An older but well-known example is the [local XSS issue in the Skype app for iOS, first identified by Phil Purviance](https://superevr.com/blog/2011/xss-in-skype-for-ios "XSS in Skype for iOS"). The Skype app failed to properly encode the name of the message sender, allowing an attacker to inject malicious JavaScript to be executed when a user views the message. In his proof-of-concept, Phil showed how to exploit the issue and steal a user's address book.
 
-### Static Analysis
+### Static Analysis Security Testing Considerations
```
Comment (sushi2k) [inline Document/0x04h-Testing-Code-Quality.md:102]:
```suggestion
### Static Analysis - Security Testing Considerations
```

Comment (sushi2k) [reply Document/0x04h-Testing-Code-Quality.md:102]: Maybe add some kind of separator or "Security Testing Considerations for Static Analysis"? Then it reads better

**Review by sushi2k — APPROVED** [review]
All good!

File: Document/0x04h-Testing-Code-Quality.md [hunk 2]
```diff
@@ -198,13 +198,13 @@ Consider how data will be rendered in a response. For example, if data is render
 
 For a comprehensive list of escaping rules and other prevention measures, refer to the [OWASP XSS Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html "OWASP XSS Prevention Cheat Sheet").
 
-### Dynamic Analysis
+### Dynamic Analysis Security Testing Considerations
```
Comment (cpholguera) [inline Document/0x04h-Testing-Code-Quality.md:201]:
```suggestion
### Dynamic Analysis - Security Testing Considerations
```

**Pull request merged**